File: base/src/test/java/com/mindera/skeletoid/threads/utils/CallbackTaskUnitTest.java
package com.mindera.skeletoid.threads.utils;

import org.junit.Test;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.times;
import static org.mockito.Mockito.verify;

public class CallbackTaskUnitTest {

    @Test
    public void testTaskCompleted() {

        Runnable runnable = mock(Runnable.class);
        CallbackTask.ICallbackTask callbackTask = mock(CallbackTask.ICallbackTask.class);

        CallbackTask task = new CallbackTask(runnable, callbackTask);
        task.run();

        verify(runnable, times(1)).run();
        verify(callbackTask, times(1)).taskComplete();
    }
}

File: base/src/test/java/com/mindera/skeletoid/threads/utils/CallbackTaskUnitTest.kt
package com.mindera.skeletoid.threads.utils

import com.mindera.skeletoid.threads.utils.CallbackTask.ICallbackTask
import org.junit.Test
import org.mockito.Mockito

class CallbackTaskUnitTest {
    @Test
    fun testTaskCompleted() {
        val runnable =
            Mockito.mock(Runnable::class.java)
        val callbackTask = Mockito.mock(ICallbackTask::class.java)
        val task = CallbackTask(runnable, callbackTask)
        task.run()
        Mockito.verify(runnable, Mockito.times(1)).run()
        Mockito.verify(callbackTask, Mockito.times(1)).taskComplete()
    }
}